internal/nmc/module_nmc_controller.go

```diff
@@ -0,0 +1,139 @@
+	return &moduleNMCHandler{
+		reconHelper: reconHelper,
+	}
+}
+
+// func (mnr *ModuleNMCReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (mnh *moduleNMCHandler) handleModuleNMC(ctx context.Context, mld *api.ModuleLoaderData, nodes []v1.Node) error {
+	var sumErr error
+	for _, node := range nodes {
+		nodeKernelVersion := strings.TrimSuffix(node.Status.NodeInfo.KernelVersion, "+")
+
+		shouldBeOnNode, err := mnh.reconHelper.shouldModuleRunOnNode(node, mld, nodeKernelVersion)
+		if err != nil {
+			sumErr = fmt.Errorf("%v: %v", sumErr, err)
+			continue
+		}
+		if shouldBeOnNode {
+			err = mnh.reconHelper.enableModuleOnNode(ctx, mld, node.Name, nodeKernelVersion)
+		} else {
+			err = mnh.reconHelper.disableModuleOnNode(ctx, mld.Namespace, mld.Name, node.Name)
+		}
+		if err != nil {
+			sumErr = fmt.Errorf("%v: %v", sumErr, err)
+		}
+	}
+
+	if sumErr != nil {
+		return fmt.Errorf("failed to handle module %s/%s with nodes: %v", mld.Namespace, mld.Name, sumErr)
+	}
+	return nil
+}
+
+//go:generate mockgen -source=module_nmc_controller.go -package=nmc -destination=mock_module_nmc_controller.go
+
+type moduleNMCHandlerHelperAPI interface {
+	shouldModuleRunOnNode(node v1.Node, mld *api.ModuleLoaderData, nodeKernelVersion string) (bool, error)
+	enableModuleOnNode(ctx context.Context, mld *api.ModuleLoaderData, nodeName, kernelVersion string) error
+	disableModuleOnNode(ctx context.Context, modNamespace, modName, nodeName string) error
+}
+
+type moduleNMCHandlerHelper struct {
+	client    client.Client
+	nmcHelper Helper
+}
+
+func newModuleNMCHandlerHelper(client client.Client, nmcHelper Helper) moduleNMCHandlerHelperAPI {
+	return &moduleNMCHandlerHelper{
+		client:    client,
+		nmcHelper: nmcHelper,
+	}
+}
+
+func (mnhh *moduleNMCHandlerHelper) shouldModuleRunOnNode(node v1.Node, mld *api.ModuleLoaderData, nodeKernelVersion string) (bool, error) {
+	if nodeKernelVersion != mld.KernelVersion {
+		return false, nil
+	}
+
+	if !utils.IsNodeSchedulable(&node) {
+		return false, nil
+	}
+
+	return utils.IsObjectSelectedByLabels(node.GetLabels(), mld.Selector)
+}
+
+func (mnhh *moduleNMCHandlerHelper) enableModuleOnNode(ctx context.Context, mld *api.ModuleLoaderData, nodeName, kernelVersion string) error {
+	logger := log.FromContext(ctx)
+	moduleConfig := kmmv1beta1.ModuleConfig{
+		KernelVersion:        kernelVersion,
+		ContainerImage:       mld.ContainerImage,
+		InTreeModuleToRemove: mld.InTreeModuleToRemove,
+		Modprobe:             mld.Modprobe,
+	}
+
+	nmc := &kmmv1beta1.NodeModulesConfig{
+		ObjectMeta: metav1.ObjectMeta{Name: nodeName},
+	}
+
+	opRes, err := controllerutil.CreateOrPatch(ctx, mnhh.client, nmc, func() error {
+		return mnhh.nmcHelper.SetNMCAsDesired(ctx, nmc, mld.Namespace, mld.Name, &moduleConfig)
+	})
+
+	if err == nil {
+		logger.Info("enable module in NMC", "name", mld.Name, "namespace", mld.Namespace, "node", nodeName, "result", opRes)
+	}
+	return err
```

Comment:
```suggestion
	if err != nil {
	        return fmt.Errorf("failed to create ..., %v, err)
	}
	
	logger.Info("enable module in NMC", "name", mld.Name, "namespace", mld.Namespace, "node", nodeName, "result", opRes)
	return nil
```

How about this instead? I think it is a bit more clear

Reply:
fixed